docs/php/symfony/task-manager/index.mdx

```diff
@@ -30,55 +36,237 @@ task_manager:
+```php
+use Torr\TaskManager\Manager\TaskManager;
+
+public function example (TaskManager $taskManager)
+{
+	$englishPimImport = new PimImportTask("en");
+	$taskManager->enqueue($englishPimImport);
+}
+```
+
+If you need to pass stamps, you can do that in the second parameter:
 
 ```php
-$taskManager->enqueue($message, "unique.message.key");
+use Symfony\Component\Messenger\Stamp\DispatchAfterCurrentBusStamp;
+
+$taskManager->enqueue(new PimImportTask("en"), [
+	new DispatchAfterCurrentBusStamp(),
+]);
+````
+
+
+:::best-practice
+When queueing tasks inside a task handler, you should always add the `DispatchAfterCurrentBusStamp`.
+:::
+
+
+You can also queue tasks via the CLI:
+
+You can automatically queue registered tasks via the CLI:
+
+```shell
+bin/console task-manager:queue
 ```
 
-But it is recommended to add a name to the task itself:
+You can either interactively select the tasks or directly pass the task IDs to the command:
+
+```shell
+bin/console task-manager:queue task-id-1 task-id-2
+```
+
+
+## Task Handlers
+
+As this bundle builds on top of Symfony messages, you define your task handler as message handler:
 
 ```php
-class ImportDataTask implements UniqueMessageInterface
+use Symfony\Component\Messenger\Attribute\AsMessageHandler;
+use Torr\TaskManager\Director\TaskDirector;
+
+final readonly class PimImportMessageHandler
 {
 	public function __construct (
-		private string $locale,
+		private TaskDirector $taskDirector,
 	) {}
 
-	/**
-	 * @inheritDoc
-	 */
-	public function getJobId () : ?string
+	#[AsMessageHandler]
+	public function onPimImport (PimImportMessage $message) : void
 	{
-		return "import-data.{$this->locale}";
+		// start the run
+		$run = $this->taskDirector->startRun($message);
+
+		// you can get the IO from the run director
+		$io = $run->getIo();
+
+		// ... do your thing ...
+
+		// at the end you mark your task as finished and indicate, whether the task was handled successfully
+		$run->finish(success: true);
 	}
 }
 ```
 
+When your task failed, you have two options: you can either throw an exception, to mark this run as failed (and let the retry functionality kick in) or you can finish handling the task, but mark it as `success: false`.
+
+You use the task director to start a run for the given task. This will return a run director, that helps you with working through your run:
+
+- you can get IO to display information on the console
+- the IO output will automatically be logged as well
+- you can mark the run as success / failure
+
 :::best-practice
-To ensure consistent job ids, you should always use the `UniqueMessageInterface` in favor of inline job keys.
+While not explicitly `->finish()`ed tasks will be handled properly, you should always `finish` all your task runs.
+:::
+
+
+## Task Log
+
+The task and run directors automatically log the output of your tasks, whether they succeeded and some metadata. This includes the task object itself, so it must be serializable.
```

Comment:
```suggestion
The `TaskDirector` and `RunDirector` automatically log the output of your tasks, whether they succeeded including some metadata. This includes the task object itself, so it must be serializable.
```

Reply:
The `including` is wrong, as the metadata is not included in the output / status. Also I don't want to duplicate "Include" in 5 words. Can you change that? The rest can be adopted.